posts/a-devs-intro-to-devops.md

```diff
@@ -0,0 +1,46 @@
+---
+authors:
+- Dennis Sharpe
+tags:
+- DevOps
+- AWS GovCloud
+- OpenShift
+- Infrastructure as Code
+- IaC
+date: 2015-04-23T12:21:50.000Z
+title: "A Dev's Intro to DevOps"
+image: 
+---
+I have recently been sucked into the field known as DevOps and I thought it would be fun to share my story. I have spent most of my career in the application development space. I am curious if others have had similar experiences.
+
+##The Project
+
+The project started with a web application that was running on OKD (open source OpenShift) with tons of open source libraries. The objective was to deploy this application in the AWS GovCloud with FedRAMP High compliance. Just in case you are wondering, this required _a lot_ of work. We had to move to several commercial products versus open source (i.e. OpenShift) and move to AWS native services wherever possible.
+
+From an application perspective, code needed to be changed to interface with the new services. From a platform perspective, the whole environment needed to be built out from scratch in the GovCloud.
+
+##Freight Train
+
+The challenge here definitely hit me like a freight train. Challenge #1 was to basically set up the network. My first thought was, “What? The network was always already there when I did my work. _We_ have to set this up?” I’m being a bit facetious but I am definitely used to worrying much more about my application architecture than network architecture. Luckily, I was on a great team with talented engineers ready to build the network.
+
+##Tools
+
+As you might expect, there are so many different tools out there to enable Infrastructure as Code (IaC). I relied on my teammates to make selections based on their experience. The primary tools we used were Terraform, Packer and Ansible. I have written applications in a variety of languages so learning these “languages” should be no sweat, right? Well, these tools/languages reminded me more of elaborate configuration files than a programming language. I hope you like YAML because it is everywhere in IaC.
+
+##Automation
+
+Now that we can create scripts to build infrastructure, we can automate the complete setup of the environment. That came in real handy when we needed to build the production environment very similarly to the staging environment. The scripts can even be triggered by a pipeline just like application development code. But who builds the pipeline infrastructure? You need a bootstrap script or two to get the environment working that will build your environment. ;)
```

Comment:
That came in handy when we needed to build the production environment similarly to the staging environment.